src/cuda/nn/smax_2D_row.cu

```diff
@@ -0,0 +1,91 @@
+#ifndef __NN_SMAX_2D_ROW_ZIG__
+#define __NN_SMAX_2D_ROW_ZIG__
+
+#include "nn_utils.cu"
+
+extern "C" void smax_2D_row_forward(
```

Comment:
Reminder to myself to add an option to swap out the `CUDNN_SOFTMAX_"ALGO"` type so we can support log softmax with this same function as well.